### DISCUSSION TITLE
instant ngp on the web

### DISCUSSION THREAD

**Discussion by Jiakui** [NVlabs/instant-ngp#494] [Q&A]
Dear author, 

I want to run instant ngp in the web browser, just the  rendering part , not including the training part.  

Can you give me some suggestion how to do it ?

Thanks

**Comment by Tom94** [comment]
I'm afraid this is quite complicated and would require much engineering effort. Short of baking the NeRF into a data structure more amenable to webgl rendering (such as [PlenOctrees](https://alexyu.net/plenoctrees/)), you'd have to re-implement the entire inference pipeline of [tiny-cuda-nn](https://github.com/nvlabs/tiny-cuda-nn) in glsl, express the trained weights of the hash encoding + neural network in terms of textures, and then run that.

**Reply by entropical** [reply]
Would it be possible to convert it to a point cloud, and then render it with something like Potree?

**Reply by Tom94** [reply]
Likely yes, although the resulting rendering quality will depend strongly on the type / quality of point cloud your algorithm will generate.

**Reply by pwais** [reply]
do you think you'll ever do a WebGPU port of tiny-cuda-nn ? ;)

**Reply by Tom94** [reply]
Not in the near future. But if support for low-level GPU primitives (like tensorcores) in webgl / webcl gets better, there might be a time!

**Reply by nosy-b** [reply]
@Tom94 Isn't there a way with the RGBA_slices? Would be pretty limited in terms of rendering quality but simple to implement in WebGL, WDYT?

**Reply by Tom94** [reply]
Sure, but I wouldn't consider this to be "instant NGP on the web". It's just a volumetric light field texture that could be obtained by any means (including other NeRF methods) -- similar to what https://phog.github.io/snerg/ or https://alexyu.net/plenoctrees/ already did (the latter even having a webgl demo).

**Reply by zba** [reply]
May be just webrtc stream ?

**Comment by Lanson426** [comment]
Hello, have you made it? I also need this technology now. Could you give me some advice
